daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -323,6 +327,24 @@ downloadPackage args pid = do
     convPid :: LF.PackageId -> L.PackageId
     convPid (LF.PackageId text) = L.PackageId $ TL.fromStrict text
 
+data RemoteDalf = RemoteDalf
+    { remoteDalfFp :: FilePath
+    , remoteDalfBs :: BS.ByteString
+    , remoteDalfIsMain :: Bool
+    }
+runLedgerGetDalfs :: LedgerFlags -> [LF.PackageId] -> IO [RemoteDalf]
+runLedgerGetDalfs lflags pkgIds = do
+    args <- getDefaultArgs lflags
+    m <- downloadAllReachablePackages args pkgIds
+    pure
+        [ RemoteDalf {..}
+        | (i :: Int, (_pid, pkg)) <- zip [0 ..] $ Map.toList m
+        , let (bsl, pid) = LFArchive.encodeArchiveAndHash pkg
+        , let remoteDalfFp = (T.unpack $ recoverPackageName pkg ("dep" <> show i, pid)) <.> "dalf"
```

Comment:
Can we default to the package id as the name intead of `dep$i`? The latter doesn’t really provide any meaningful info.

Reply:
sure, I choose the `dep` prefix in order to be aligned on how the `fetch-dar` command names the dalfs. should we change that as well?k

Reply:
yeah I think I’d change both

Reply:
ok, done